client/app.jsx

```diff
@@ -10,7 +11,12 @@ export default function App() {
 
     <Routes>
       <Route path="/" element={<Home />} />
+      <Route path="/products/:productId" element={<ProductViewWithParams />} />
     </Routes>
 
   );
 }
+function ProductViewWithParams() {
```

Comment:
I like this approach. It's how I've done it. That way my component isn't burdened with the routing choice I make.